File: clear-database.js
#!/usr/bin/env node
/**
 * QSecure Radar — Database Cleanup Script
 * Clears all scan, asset, and vulnerability data from Supabase
 */

const { createClient } = require('@supabase/supabase-js');

const SUPABASE_URL = 'https://shinmrlkbaggbwpzhlcl.supabase.co';
const SUPABASE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InNoaW5tcmxrYmFnZ2J3cHpobGNsIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzQxMjQ5OTksImV4cCI6MjA4OTcwMDk5OX0.xYRrKNOdUD3g-APqEGSx9yG6qg6YpCeziIGY-gqPYhA';

async function clearDatabase() {
  try {
    // Initialize Supabase client
    const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);
    
    console.log('\n🔄 Connecting to Supabase...');
    console.log(`   Project: ${SUPABASE_URL}\n`);
    
    // Tables to clear (in order of FK dependencies)
    const tablesToClear = [
      'scan_history',
      'reports',
      'audit_log',
      'cbom',
      'cyber_rating',
      'pqc_scores',
      'software',
      'nameservers',
      'crypto_overview',
      'ip_subnets',
      'ssl_certs',
      'domains',
      'assets'
    ];
    
    console.log('🗑️  Clearing tables...');
    
    for (const table of tablesToClear) {
      try {
        const { error } = await supabase
          .from(table)
          .delete()
          .neq('id', '00000000-0000-0000-0000-000000000000');
        
        if (error) {
          console.log(`   ⚠️  ${table}: ${error.message}`);
        } else {
          console.log(`   ✅ ${table}`);
        }
      } catch (e) {
        console.log(`   ⚠️  ${table}: ${e.message}`);
      }
    }
    
    console.log('\n📊 Verifying tables are empty...\n');
    
    // Verify each table is empty
    for (const table of tablesToClear) {
      try {
        const { count, error } = await supabase
          .from(table)
          .select('*', { count: 'exact', head: true });
        
        const rowCount = count || 0;
        const status = rowCount === 0 ? '✅ EMPTY' : `⚠️  ${rowCount} rows`;
        console.log(`   ${table.padEnd(20)} ${status}`);
      } catch (e) {
        console.log(`   ${table.padEnd(20)} ⚠️  Could not verify`);
      }
    }
    
    console.log('\n' + '='.repeat(60));
    console.log('✨ Database cleared successfully!');
    console.log('='.repeat(60));
    console.log('\n📋 Next steps:');
    console.log('   1. Run supabase/fresh-seed.sql to load sample assets');
    console.log('   2. Refresh dashboard to see fresh asset inventory');
    console.log('   3. Start scanning with Unified Scanner\n');
    
    process.exit(0);
  } catch (error) {
    console.error('\n❌ Error:', error.message);
    process.exit(1);
  }
}

// Run
clearDatabase();

File: clear_database.py
#!/usr/bin/env python3
"""
QSecure Radar — Database Cleanup Script
Clears all scan, asset, and vulnerability data from Supabase
"""

import os
import sys
from supabase import create_client

# Supabase credentials (from config)
SUPABASE_URL = "https://shinmrlkbaggbwpzhlcl.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InNoaW5tcmxrYmFnZ2J3cHpobGNsIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzQxMjQ5OTksImV4cCI6MjA4OTcwMDk5OX0.xYRrKNOdUD3g-APqEGSx9yG6qg6YpCeziIGY-gqPYhA"

# Read SQL commands


def read_sql_file(filename):
    with open(filename, 'r') as f:
        return f.read()


def clear_database():
    try:
        # Initialize Supabase client
        supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

        print("🔄 Connecting to Supabase...")
        print(f"   Project: {SUPABASE_URL}\n")

        # Tables to clear (in correct order due to FK constraints)
        tables_to_clear = [
            'scan_history',
            'reports',
            'audit_log',
            'cbom',
            'cyber_rating',
            'pqc_scores',
            'software',
            'nameservers',
            'crypto_overview',
            'ip_subnets',
            'ssl_certs',
            'domains',
            'assets'
        ]

        print("🗑️  Clearing tables...")
        for table in tables_to_clear:
            try:
                supabase.table(table).delete().neq(
                    'id', '00000000-0000-0000-0000-000000000000').execute()
                print(f"   ✅ {table}")
            except Exception as e:
                print(f"   ⚠️  {table}: {str(e)}")

        print("\n📊 Verifying tables are empty...\n")

        # Verify each table is empty
        for table in tables_to_clear:
            try:
                response = supabase.table(table).select(
                    'COUNT()', count='exact').execute()
                count = response.count if hasattr(response, 'count') else 0
                status = "✅ EMPTY" if count == 0 else f"⚠️  {count} rows remain"
                print(f"   {table:<20} {status}")
            except Exception as e:
                print(f"   {table:<20} ⚠️  Could not verify")

        print("\n" + "="*60)
        print("✨ Database cleared successfully!")
        print("="*60)
        print("\n📋 Next steps:")
        print("   1. Run supabase/fresh-seed.sql to load sample assets")
        print("   2. Refresh dashboard to see fresh asset inventory")
        print("   3. Start scanning with Unified Scanner\n")

    except Exception as e:
        print(f"\n❌ Error: {e}")
        sys.exit(1)


if __name__ == '__main__':
    clear_database()
